libs/designer/src/lib/ui/panel/recommendation/recommendationPanelContext.tsx

```diff
@@ -49,23 +49,36 @@ export const RecommendationPanelContext = (props: RecommendationPanelContextProp
 
   const [selectionState, setSelectionState] = useState<SelectionState>(SELECTION_STATES.SEARCH);
 
-  const selectedOperationId = useSelectedSearchOperationId();
   const { data: allOperations, isLoading: isLoadingOperations } = useAllOperations();
-  const selectedOperation = allOperations.find((o) => o.id === selectedOperationId);
+  const [selectedOperation, setSelectedOperation] = useState<DiscoveryOperation<DiscoveryResultTypes> | undefined>(undefined);
+  const [isWaitingOperationsGroup, setIsWaitingOperationGroup] = useState<boolean>(false);
 
   const selectedOperationGroupId = useSelectedSearchOperationGroupId();
   const { data: allConnectors } = useAllConnectors();
   const selectedConnector = allConnectors?.find((c) => c.id === selectedOperationGroupId);
 
+  // effect to set the current list of operations by group
   useEffect(() => {
-    if (!allOperations || !selectedOperationGroupId) return;
-    const filteredOps = allOperations.filter((operation) => {
-      const apiId = operation.properties.api.id;
-      return areApiIdsEqual(apiId, selectedOperationGroupId);
+    if (!selectedOperationGroupId) return;
+
+    const searchOperation = SearchService().getOperationsByConnector?.bind(SearchService());
+
+    const searchResultPromise = searchOperation
+      ? searchOperation(selectedOperationGroupId, filters['actionType']?.toLowerCase())
+      : Promise.resolve(
+          (allOperations ?? []).filter((operation) => {
+            const apiId = operation.properties.api.id;
+            return areApiIdsEqual(apiId, selectedOperationGroupId);
+          })
+        );
+
+    searchResultPromise.then((filteredOps) => {
+      setAllOperationsForGroup(filteredOps);
+      setIsWaitingOperationGroup(false);
     });
-    setAllOperationsForGroup(filteredOps);
     setSelectionState(SELECTION_STATES.DETAILS);
-  }, [selectedOperationGroupId, allOperations]);
+    setIsWaitingOperationGroup(true);
```

Comment:
This might cause a race condition for `isWaitingOperationsGroup` to remain set to true if the search results promise finishes too quick, can you move the `setIsWaitingOperationGroup(true)` up above the `searchResultPromise` just to be safe, and also add a catch in case the promise fails that also resets the `isWaitingOperationsGroup`

Also just a nit but can you rename the state variable to `isLoadingOperationGroup` instead

Reply:
Thanks for the comment Riley. Like we agreed offline, the race condition isn't something to worry about given the single threaded nature of JS. I moved it up for readability, did the rename and the `finally`